src/Serializer.php

```diff
@@ -0,0 +1,52 @@
+<?php
+/**
+ * This file is part of the prooph/pdo-snapshot-store.
+ * (c) 2016-2017 prooph software GmbH <contact@prooph.de>
+ * (c) 2016-2017 Sascha-Oliver Prolic <saschaprolic@googlemail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Prooph\SnapshotStore\Pdo;
+
+final class DefaultSerializerStrategy implements SerializerStrategy
+{
+    /**
+     * callable
+     */
+    private $serializer = 'serialize';
+
+    /**
+     * callable
+     */
+    private $deserialize = 'unserialize';
+
+    public function __construct(?callable $serializer, ?callable $deserialize)
+    {
+        if ($serializer && $deserialize) {
```

Comment:
`if (null !== $serializeCallback && null !== $unserializeCallback)` is much faster.

see https://3v4l.org/QaoTI